src/context.rs

```diff
@@ -1274,9 +1274,83 @@ impl Context {
 
     /// Extends a PCR with the specified digests.
     ///
+    /// # Arguments
+    /// *`pcr_handle`- A [PcrHandle] to the PCR slot that is to be extended.
+    /// *`digests` - The [DigestValues] with which the slot shall be extended.
+    ///
     /// # Details
     /// This method is used to cause an update to the indicated PCR. The digests param
     /// contains the digests for specific algorithms that are to be used.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use tss_esapi::{
+    /// #     Context, Tcti,
+    /// #     constants::{
+    /// #         algorithm::Cipher,
+    /// #         types::session::SessionType,
+    /// #     },
+    /// #     session::SessionAttributesBuilder,
+    /// #     handles::PcrHandle,
+    /// # };
+    /// # use std::{env, str::FromStr};
+    /// # // Create TCTI that uses the correct software for testing
+    /// # let tcti = env::var("TEST_TCTI")
+    /// #   .map(|env_tcti_str| Tcti::from_str(&env_tcti_str).expect("Failed to parse TCTI string."))
+    /// #   .unwrap_or_else(|_| Tcti::Mssim(Default::default()));
+    /// # // Create context
+    /// # let mut context = unsafe {
+    /// #     Context::new(tcti).expect("Failed to create Context")
+    /// # };
+    /// # // Create session for a pcr
+    /// # let pcr_session = context
+    /// #     .start_auth_session(
+    /// #         None,
+    /// #         None,
+    /// #         None,
+    /// #         SessionType::Hmac,
+    /// #         Cipher::aes_256_cfb(),
+    /// #         tss_esapi::constants::algorithm::HashingAlgorithm::Sha256,
+    /// #     )
+    /// #     .expect("Failed to create session")
+    /// #     .expect("Recived invalid handle");
+    /// # let (session_attributes, session_attributes_mask) = SessionAttributesBuilder::new()
+    /// #     .with_decrypt(true)
+    /// #     .with_encrypt(true)
+    /// #     .build();
+    /// # context.tr_sess_set_attributes(pcr_session, session_attributes, session_attributes_mask)
+    /// #     .expect("Failed to set attributes on session");
+    ///
+    /// use std::convert::TryFrom;
+    /// use tss_esapi::{
+    ///     structures::{DigestValues, Digest},
+    ///     constants::algorithm::HashingAlgorithm,
+    /// };
+    /// // Extend both sha256 and sha1
+    /// let mut vals = DigestValues::new();
+    /// vals.set(
+    ///     HashingAlgorithm::Sha1,
+    ///     Digest::try_from(vec![
+    ///         1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
+    ///     ])
+    ///     .expect("Failed to create Digest from data"),
```

Comment:
I'm wondering if this could be hidden into a variable as well, and just used as `digest` - kind of like `pcr_session`

Reply:
Fixed